### PR TITLE
Add deprecation message to CallDefault function partly resolves issue#6257

### LINKED ISSUE #6257: Deprecate/Remove `CallDefault`
## 🧰 Task

In `napari/utils/misc.py`, `CallDefault` seem to not be used since #2266. We should investigate adding a deprecation message and thinking about removal if this is not used anywhere else.

### PR DESCRIPTION
**Pull request by MarchisLost**
# References and relevant issues

Partly closes #6257 by adding a deprecation message to the CallDefault function that seems to not be used since https://github.com/napari/napari/pull/2266

# Description

- [X] - My PR is the minimum possible work for the desired functionality
- [ ] - I have commented my code, particularly in hard-to-understand areas
- [ ] - I have made corresponding changes to docstrings and documentation (open a PR on the docs repository (https://github.com/napari/docs) if relevant!)
- [ ] - I have added tests that prove my fix is effective or that my feature works
- [X] - If I included new strings, I have used `trans._("some string")` to make them localizable.
  (For more information see our [translations guide](https://napari.org/developers/translations.html)).

### CODE REVIEW

**Comment by melissawm** [comment]
Hi @MarchisLost  - thanks for your contribution!

Would you mind changing the title of your PR to be more descriptive of the problem you are solving? This will help your PR get more visibility and will help the maintainers understand your work better. In your case, I would suggest something like "Add deprecation message to CallDefault function" or something similar.

Welcome and let us know if you have questions 😄

**Comment by MarchisLost** [comment]
Hi, thank you for the feedback.
I have changed the PR title.

**Comment by DragaDoncila** [comment]
Hi @MarchisLost thanks for the PR! I'm pinging @melonora for review, because I think this is similar to #6836 that he recently looked at

File: napari/utils/misc.py [hunk 1]
```diff
@@ -398,6 +398,13 @@ def abspath_or_url(relpath: T, *, must_exist: bool = False) -> T:
 
 
 class CallDefault(inspect.Parameter):
+    warnings.warn(
+        trans._(
+            '`CallDefault` is deprecated since 0.4.6 and will be removed in the future',
+        ),
+        category=DeprecationWarning,
+    )
```
Comment (melonora) [inline napari/utils/misc.py]:
```suggestion
    warnings.warn(
        trans._(
            '`CallDefault` is deprecated in napari v0.5.0 and will be removed in v0.6.0.',
        ),
        category=DeprecationWarning,
    )
```

**Review by melonora — CHANGES_REQUESTED** [review]
Thanks @MarchisLost (interesting name haha) and welcome to napari! I have checked and indeed this does not seem to be used at all. However, for good practice we have 1 release always for warning that it is deprecated and then the release after that we remove. If we slightly change the message we are good to go for merging this one. Once you have accepted the suggestion I will mark your PR ready for merge and then if nobody else has obligations we merge within a day.

**Review by melonora — APPROVED** [review]

**Comment by melonora** [comment]
Approved! Thanks:)

**Comment by MarchisLost** [comment]
Hi @melonora thank you!

**Comment by melonora** [comment]
in it goes!

**Pull request merged**